core/src/test/java/org/mule/el/context/MessageAttachmentsTestCase.java

```diff
@@ -135,8 +135,9 @@ public void inboundKeySet() throws Exception
         DefaultMuleMessage message = new DefaultMuleMessage("", muleContext);
         message.addInboundAttachment("foo", Mockito.mock(DataHandler.class));
         message.addInboundAttachment("bar", Mockito.mock(DataHandler.class));
-        assertEquals("foo", evaluate("message.inboundAttachments.keySet().toArray()[0]", message));
-        assertEquals("bar", evaluate("message.inboundAttachments.keySet().toArray()[1]", message));
+        assertEquals(2, evaluate("message.inboundAttachments.keySet().size()", message));
```

Comment:
why is this test case affected by removing asm? this is a smell that something got broken